Civilian bounties & HAUL Gauntlets 2: Economic Boogaloo

KoboldCommando opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

# About The Pull Request

Big big thanks to DoctorSquishy and Borbop for helping squash the bug holding this in limbo, and figure out working with the split-up signals!

This PR is back from the dead, and finally in working form that shouldn't break pirates!

----------------

Ports cargo stuff from tg! Including the H.A.U.L. Gauntlets, and the civilian bounty machines and system.

These things can be researched (data theory, and mining technology), and then printed at the cargo techfab. The machines can be assembled and disassembled as normal.

Cargo bounties work just like tg. You plug in your ID, grab your bounty, send it via the pad, and have cargo export the resulting bounty cube. They can tag it with an export scanner for a tip!

Export scanners also no longer need to be linked to a console to work (it was pointless).

HAUL Gauntlets do not have an equip/unequip time. I'm not sure what the reason for this was exactly and it was a pain to get working before Bee gets a deeper equip/unequip system rework.

An inserted ID can potentially wind up "stuck" in the computer if the pad gets disabled. The computer has a line in the description reminding people that they can manually remove it with an alt-click.

The prices are not adjusted yet, and it pulls from the same list of bounties as the existing ones. So it will likely need balance tweaks going forward. But we can look into that once it's in place!

## Why It's Good For The Game

More cargo shit! Also a system that encourages cargo to interact with the station more, and the gloves help that interaction by assuring crate safety.

Also a step toward a better economy! After the early round, when early bounties have been knocked out and cargo starts to coast, they will now likely have access to the research to get this started, which gives cargo more to do and will pick up the ability for crew members to continue making money.

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog

🆑
add: Haul gauntlets can be researched and printed at cargo techfabs
add: Civilian bounty computers and pads can also be researched and printed at cargo
add: The civilian bounty system and bounty cubes is ported from tg. Get to making that cash, spacemen!
tweak: export scanners no longer need to sync with a console
/🆑

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

nednaZ commented on code/modules/cargo/export_scanner.dm:
```diff
@@ -9,37 +9,47 @@
-		to_chat(user, "<span class='warning'>You must link [src] to a cargo console first!</span>")
+	// Before you fix it:
+	// yes, checking manifests is a part of intended functionality.
+	var/datum/export_report/ex = export_item_and_contents(O, dry_run=TRUE)
```
Needs a better variable name

nednaZ commented on code/modules/cargo/export_scanner.dm:
```diff
@@ -9,37 +9,47 @@
+	// yes, checking manifests is a part of intended functionality.
+	var/datum/export_report/ex = export_item_and_contents(O, dry_run=TRUE)
+	var/price = 0
+	for(var/x in ex.total_amount)
```
Needs a better variable name

nednaZ commented on code/modules/cargo/export_scanner.dm:
```diff
@@ -9,37 +9,47 @@
+		to_chat(user, "<span class='warning'>Scanned [O], no export value.")
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/scan_human = user
```
scanning_user

nednaZ commented on code/modules/cargo/exports/sheets.dm:
```diff
@@ -28,13 +27,13 @@
 
 /datum/export/stack/skin/cat
```
Does this actually still exist?
From cats, or felinids?

KoboldCommando replied:
I'm not sure if it's obtainable, but the item does still exist in the code.  Maybe you can skin Runtime somehow?  I'm not sure.  Probably better to leave it here just in case until/unless someone specifically removes the item, I think.

nednaZ reviewed (commented):
We already know it worked in the past, sans the pirate pad.
So as long as it works, we can start actually testing this again.

nednaZ commented:
Test Merging

nednaZ reviewed (changes requested):
Can you add the civilian bounty systems to maps?
Will make testing far easier

KoboldCommando commented:
Added a computer and pad to each map, in the cargo lobby apart from a couple cramped maps where it had to be in the office.  Also fixed all the mentioned issues with pointless inclusions and variable names!

nednaZ commented:
Will run through on our next set of test merges

nednaZ reviewed (approved):
Pirate pad fixed.
System appears to work as intended.

Next step, rebalancing the existing bounties!
